mozmeao/birdbox
# Ensure YouTube video embeds don't set tracking cookies

## Description

We currently have configured Birdbox to use Oembed to embed YouTube and Vimeo videos into a page, via a custom `VideoBlock`. Vimeo does not set tracking cookies, but the default YouTube embed does. 

This changeset updates our YouTube embedding to use a special embed 'finder' that rewrites the oembed markup:

* instead of youtube.com we use youtube-nocookie.com
* we use trimmed back markup that we control for the iframe, so that only essential attributes are included

- [X] I have manually tested this.

### Issue

Resolves #260

### Testing and points to consider

* Check out this branch
* Edit a page that allows adding a VideoBlock (eg a `Blog Post` or a `General Purpose Page`) 
* Add a link to a YT video you have not embedded before (because Wagtail caches old embed data in the database)
* Publish the page and view it, checking the video works and can be played in fullscreen
* In devtools: 
  * confirm that the video is loaded from the youtube-nocookie.com domain
  * confirm no cookie (or a pending consent cookie) is all that is set from that domain
* Do you agree that no <p> tag is needed inside the `iframe` (which is what we do on mozorg - see https://github.com/mozilla/bedrock/issues/14042)? If not, can you explain why it's there as it triggers HTML validation failure.
